src/types/__test__/article.test.ts

```diff
@@ -663,6 +665,51 @@ describe('edit article', () => {
     )
     expect(_get(result5, 'data.article.replyToDonator')).toBe(replyToDonator)
   })
+  test('edit comment settings', async () => {
+    const server = await testClient({
+      isAuth: true,
+      isAdmin: false,
+    })
+    const result = await server.executeOperation({
+      query: GET_ARTICLE,
+      variables: {
+        input: {
+          mediaHash,
+        },
+      },
+    })
+    expect(_get(result, 'data.article.canComment')).toBeTruthy()
+
+    // can not turn off
+    const result2 = await server.executeOperation({
+      query: EDIT_ARTICLE,
+      variables: {
+        input: {
+          id: ARTICLE_ID,
+          canComment: false,
+        },
+      },
+    })
+    expect(result2.errors).not.toBeUndefined()
+
+    // can turn on
+    const atomService = new AtomService()
+    await atomService.update({
+      table: 'draft',
+      where: { id: 1 },
+      data: { canComment: false },
+    })
+    const result3 = await server.executeOperation({
+      query: EDIT_ARTICLE,
+      variables: {
+        input: {
+          id: ARTICLE_ID,
+          canComment: true,
+        },
+      },
+    })
+    expect(_get(result3, 'data.editArticle.canComment')).toBeTruthy()
+  })
```

Comment:
need a test for "turn off" to "turn on"?

Reply:
result3 test is a test for "turn off" to "turn on"？

Reply:
shall we make the "off" article via API instead of updating db directly (L697)?